packages/orchestrator-ui-components/src/hooks/useQueryWithFetch.ts

```diff
@@ -0,0 +1,37 @@
+import { useQuery } from 'react-query';
+import { Variables } from 'graphql-request/build/cjs/types';
+import { signOut } from 'next-auth/react';
+import { useSessionWithToken } from './useSessionWithToken';
+
+export const useQueryWithFetch = <T, V extends Variables>(
+    url: string,
+    queryVars: V,
+    queryKey: string,
+) => {
+    const { session } = useSessionWithToken();
+    let requestHeaders = {};
+
+    if (session) {
+        const { accessToken } = session;
+
+        requestHeaders = {
+            authorization: `Bearer ${accessToken}`,
+        };
+    }
```

Comment:
This can be extracted in a reusable function, something like getRequestHeaders or getAuthorizationHeaders, to apply in both useQueryWithFetch and useQueryWithGraphql. As a bonus you won't need to use "let"